css/card.css

```diff
@@ -389,25 +389,94 @@ section#gacha-controls fieldset:nth-child(2) > input:first-child {
   display: none;
 }
 
-@media screen and (max-width: 1400px) {
-  #gacha-controls {
-    /* make the card smaller when the screen is resized */
+.dialog-container,
+.dialog-overlay {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 5; /* set to 10 so that it's greater than nav a.active's z-index */
+}
+
+/* create a clearer separation between the different text sections of the card */
+.dialog-container h3:not(:first-child) {
+  border-top: 1px solid rgba(0, 0, 0, 0.1);
+  padding-top: 1em;
+}
+
+.dialog-overlay {
+  background-color: rgba(43, 46, 56, 0.7);
+}
+
+/* any width but 770px height or less */
+@media screen and (max-height: 770px) {
+  #card-details-dialog {
+    /* let whole dialog scroll */
+    overflow: auto;
+  }
+  .details-dialog-card {
+    /* make the card sit up the top instead */
+    align-self: start;
+  }
+  .details-dialog-text {
+    /* Override overflow: auto for small screens, 
+    and let whole dialog scroll instead of just the text section */
+    overflow: visible;
+    height: max-content;
+  }
+}
+
+/* only make card bigger if the screen height is tall enough to support that card height in full */
+@media screen and (min-width: 1400px) and (min-height: 950px) {
+  .details-dialog-card {
+    /* make the card larger when the screen is resized - 500px width end state*/
+    --card-width: 250px;
+    --card-height: calc(250px * 1.395);
+  }
+}
+
+/* only make card bigger if the screen height is tall enough to support that card height in full */
+@media screen and (min-width: 1600px) and (min-height: 1070px) {
+  .details-dialog-card {
+    /* make the card larger when the screen is resized - 600px width end state*/
     --card-width: 300px;
-    --card-height: 419px;
+    --card-height: calc(300px * 1.395);
   }
+}
 
+/* only make card bigger if the screen height is tall enough to support that card height in full */
+@media screen and (min-width: 2200px) and (min-height: 1350px) {
   .details-dialog-card {
-    width: 25vw;
-    height: calc(25vw * 1.395);
+    /* make the card larger when the screen is resized - 700px width end state*/
+    --card-width: 350px;
+    --card-height: calc(350px * 1.395);
+  }
+  /* adjust the background grain overlay so it doesn't look so pixellated */
+  .details-dialog-card.basic:after, #gacha-controls .card-image.basic:after {
+    background-size: 120% !important;
+    background-repeat: repeat;
   }
+}
 
+@media screen and (max-width: 1400px) {
+  #gacha-controls {
+    /* make the card smaller when the screen is resized */
+    --card-width: 300px;
+    --card-height: 419px;
```

Comment:
(nit) is this 419 height 300px * 1.395? express it like calc to be consistent with the rest of the rules? else explain the slightly different math

Reply:
The strange thing with the card sizing is that it's all based off the original 1488 × 2079 card exports (1.397 ratio between the two dimensions), but as you get smaller you start calculating fractions of pixels. If you get the dimensions from a card resize in Photoshop for example, you get 300x419 rather than 300x419.15. Will take a look at cleaning this up.